File: src/main/java/com/todolist/gateway/service/RouterValidator.java
package com.todolist.gateway.service;

import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class RouterValidator {

  public static final List<String> whiteList = List.of(
      "^/api/admin/login$",
      "^/api/member/login$"
  );

  public boolean needAuth(String requestPath) {
    return whiteList.stream().noneMatch(requestPath::matches);
  }

}

File: src/main/java/com/todolist/gateway/service/RouterValidator.kt
package com.todolist.gateway.service

import org.springframework.stereotype.Component

@Component
class RouterValidator {
    fun needAuth(requestPath: String): Boolean {
        return whiteList.stream().noneMatch { regex: String ->
            requestPath.matches(
                regex.toRegex()
            )
        }
    }

    companion object {
        val whiteList = listOf(
            "^/admin/login$",
            "^/member/login$"
        )
    }
}
